suite-native/module-send/src/components/CoinControl/SwitchCoinControlButton.tsx

```diff
@@ -0,0 +1,67 @@
+import { useNavigation } from '@react-navigation/native';
+
+import { Box, Button, Text } from '@suite-native/atoms';
+import { Icon } from '@suite-native/icons';
+import { Translation } from '@suite-native/intl';
+import {
+    RootStackParamList,
+    SendStackParamList,
+    SendStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+
+import { useUtxoSelection } from '../../hooks/useUxtoSelection';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    SendStackParamList,
+    SendStackRoutes.SendUtxo,
+    RootStackParamList
+>;
+
+type SwitchCoinControlButtonProps = {
+    accountKey: string;
+    amount?: string;
+};
+
+export const SwitchCoinControlButton = ({ accountKey, amount }: SwitchCoinControlButtonProps) => {
+    const { selectedUtxos, totalSelectedAmount, isCoinControlEnabled } = useUtxoSelection();
+    const navigation = useNavigation<NavigationProps>();
+
+    const isMissingUtxos =
+        isCoinControlEnabled && amount && Number(totalSelectedAmount) < Number(amount);
+
+    const openCoinControlScreen = () => {
+        navigation.navigate(SendStackRoutes.SendUtxo, {
+            accountKey,
+            amount: amount ?? undefined,
+        });
+    };
+
+    let colorScheme: 'yellowBold' | 'primary' | 'tertiaryElevation0' = 'yellowBold';
+    if (isCoinControlEnabled) {
+        colorScheme = isMissingUtxos ? 'yellowBold' : 'primary';
+    } else {
+        colorScheme = 'tertiaryElevation0';
+    }
```

Comment:
super nitpicky, feel free to ignore completely:
I prefer to avoid using `let` unless it’s for performance optimization. From my perspective, defining `getColorScheme` function can be easier to read